packages/aws-cdk-lib/aws-lambda-nodejs/lib/function.ts

```diff
@@ -116,9 +120,16 @@ export class NodejsFunction extends lambda.Function {
       handler: handler.indexOf('.') !== -1 ? `${handler}` : `index.${handler}`,
     });
 
-    // Enable connection reuse for aws-sdk
-    if (props.awsSdkConnectionReuse ?? true) {
-      this.addEnvironment('AWS_NODEJS_CONNECTION_REUSE_ENABLED', '1', { removeInEdge: true });
+    // Enable connection reuse for aws-sdk v2, do not set for sdk v3
+    if (isSdkV2Runtime(runtime)) {
+      if (props.awsSdkConnectionReuse ?? true) {
+        this.addEnvironment('AWS_NODEJS_CONNECTION_REUSE_ENABLED', '1', { removeInEdge: true });
+      }
+    } else {
+      if (props.awsSdkConnectionReuse) {
+        Annotations.of(scope).addWarningV2('aws-cdk-lib/aws-lambda-nodejs:unusedSdkEvironmentVariable', 'The AWS_NODEJS_CONNECTION_REUSE_ENABLED environment variable does not exist in SDK v3. You have explicitly set `awsSdkConnectionReuse`; please make sure this is intentional.');
+        this.addEnvironment('AWS_NODEJS_CONNECTION_REUSE_ENABLED', '1', { removeInEdge: true });
```

Comment:
Would this env variable do anything in sdk v3? Is this for backward compatibility?

Reply:
I left this so users can set it if need be (they can also configure their environment variables directly). I figured some users may define >= Node 18 runtimes, but bundle the sdk v2 into their v2 handlers, and this would be for ease of use / backwards compatibility.